src/pages/commits/ActiveCommits/buildVariantCard/WaterfallTaskStatusIcon.test.tsx

```diff
@@ -0,0 +1,103 @@
+import { MockedProvider } from "@apollo/client/testing";
+import userEvent from "@testing-library/user-event";
+import { MemoryRouter } from "react-router-dom";
+import { GET_FAILED_TASK_STATUS_ICON_TOOLTIP } from "gql/queries";
+import { act, render, waitFor } from "test_utils/test-utils";
+import { WaterfallTaskStatusIcon } from "./WaterfallTaskStatusIcon";
+
+const props = {
+  displayName: "multiversion",
+  timeTaken: 2754729,
+  taskId: "task",
+};
+// Skip this test for now since the MockProvider mock fails to return the query data after hovering.
+// The same MockProvider mock works as expected in the corresponding Storybook file.
+test.skip("Tooltip should contain task name, duration and list of failing test names", async () => {
+  const { queryByDataCy, queryByText } = render(
+    <MemoryRouter>
+      <MockedProvider mocks={[getTooltipQueryMock]} addTypename={false}>
+        <WaterfallTaskStatusIcon {...props} status="failed" />
+      </MockedProvider>
+    </MemoryRouter>
+  );
+  userEvent.hover(queryByDataCy("failed-task-status-icon"));
+  await act(() => new Promise((resolve) => setTimeout(resolve, 0)));
+  await act(() => new Promise((resolve) => setTimeout(resolve, 0)));
+  await act(() => new Promise((resolve) => setTimeout(resolve, 0)));
+  await act(() => new Promise((resolve) => setTimeout(resolve, 0)));
+  await act(() => new Promise((resolve) => setTimeout(resolve, 0)));
+  await act(() => new Promise((resolve) => setTimeout(resolve, 0)));
+  await waitFor(() => {
+    expect(
+      queryByDataCy("failed-task-status-icon-tooltip")
+    ).toBeInTheDocument();
+  });
+  await waitFor(() => {
+    expect(queryByDataCy("failed-task-status-icon-tooltip")).toBeVisible();
+  });
+  await waitFor(() => {
+    expect(queryByText("multiversion - 45m 54s")).toBeVisible();
+  });
+
+  await waitFor(() => {
+    expect(
+      queryByText("jstests/multiVersion/remove_invalid_index_options.js")
```

Comment:
This does not exist in the mocked query response and will always fail. Replace this check with a valid value.